edward/inferences/renyi_divergence.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import six
+import numpy as np
+import tensorflow as tf
+
+from edward.inferences.variational_inference import VariationalInference
+from edward.models import RandomVariable
+from edward.util import copy
+
+try:
+    from edward.models import Normal
+    from tensorflow.contrib.distributions import kl_divergence
+except Exception as e:
+    raise ImportError(
+        "{0}. Your TensorFlow version is not supported.".format(e))
+
+
+class Renyi_divergence(VariationalInference):
+    """Variational inference with the Renyi divergence
+
+    $ \text{D}_{R}^{(\alpha)}(q(z)||p(z \mid x))
+        = \frac{1}{\alpha-1} \log \int q(z)^{\alpha} p(z \mid x)^{1-\alpha} dz $
+
+    To perform the optimization, this class uses the techniques from
+    Renyi Divergence Variational Inference (Y. Li & al, 2016)
+
+    # Notes:
+        - Renyi divergence does not have any analytic version.
+        - Renyi divergence does not have any version for non reparametrizable
```

Comment:
It does but the gradient estimator in `@li2016variational` doesn't.  I recommend just stating that this inference algorithm is restricted to variational approximations whose random variables all satisfy `rv.reparameterization_type == tf.contrib.distributions.FULLY_REPARAMETERIZED`.

Also, instead of checking this during `build_loss_and_gradients` I recommend checking this during the `__init__`. This sort of check is done statically any graph construction similar to how we check for compatible shapes in all latent variables and data during `__init__`.